packages/core/src/StateNode.ts

```diff
@@ -147,7 +149,8 @@ class StateNode<
   TContext = any,
   TStateSchema extends StateSchema = any,
   TEvent extends EventObject = EventObject,
-  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
+  TTypestate extends Typestate<TContext> = { value: any; context: TContext },
+  TResolvedTypesMeta = TypegenDisabled
```

Comment:
Adjusting those classes to support more generics is a real chore. I wonder if something can be done about this in the future. It seems that exposing a public overload and an internal implementation type is possible but I don't quite understand the assignability rules between those two. It would be worth experimenting with this in the future.